File: Design_Pattern_and_Principles/SingletonPatternExample/SingletonTest.java

public class SingletonTest {
    public static void main(String[] args) {
        Logger log1 = Logger.getInstance();
        Logger log2 = Logger.getInstance();

        log1.log("Checking singleton behavior...");

        if (log1 == log2) {
            System.out.println("Verification Passed: Both variables reference the exact same instance.");
        } else {
            System.out.println("Verification Failed: Instances do not match.");
        }
    }
}

File: Week_1/Design_Pattern_and_Principles/SingletonPatternExample/SingletonTest.cs
using System;

public class SingletonTest {
    public static void Main(string[] args) {
        Logger log1 = Logger.GetInstance();
        Logger log2 = Logger.GetInstance();

        log1.Log("Checking singleton behavior...");

        if (object.ReferenceEquals(log1, log2)) {
            Console.WriteLine("Verification Passed: Both variables reference the exact same instance.");
        } else {
            Console.WriteLine("Verification Failed: Instances do not match.");
        }
    }
}